src/DacpacTool/ScriptParser.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Reflection;
+using Microsoft.SqlTools.ServiceLayer.BatchParser;
+using Microsoft.SqlTools.Extensibility;
+
+namespace MSBuild.Sdk.SqlProj.DacpacTool
+{
+    public class ScriptParser : ICommandHandler
+    {
+        private readonly Parser _parser;
+        private bool _parsed;
+        private readonly List<string> _includedFileNames = new List<string>();
+        private StringBuilder _scriptBuilder = new StringBuilder();
+
+        public ScriptParser(string sourceFile, IVariableResolver variableResolver)
+        {
+            _parser = new Parser(this, variableResolver, new StreamReader(sourceFile), sourceFile);
+        }
+
+        private void Parse()
+        {
+            if (! _parsed)
+            {
+                _parser.Parse();
+                _parsed = true;
+            }
+        }
+
+        public List<string> CollectFileNames()
+        {
+            Parse();
+            return _includedFileNames;
+        }
+
+        public string GenerateScript()
+        {
+            Parse();
+            return _scriptBuilder.ToString();
+        }
+
+        BatchParserAction ICommandHandler.Go(TextBlock batch, int repeatCount, SqlCmdCommand tokenType)
+        {
+            batch.GetText(true, out string batchText, out LineInfo lineInfo);
+            _scriptBuilder.Append(batchText);
+            _scriptBuilder.Append("\nGO\n");
```

Comment:
This is a bit of a magic string of course. Doesn't the included library have a constant for this?

Reply:
There's one in another assembly, but it seems like a pretty small thing to add another assembly reference for. I could just create my own constant for it if you'd like, but the string "GO" was what we had here before.

Reply:
Yeah, adding an assembly reference for just a constant doesn't make much sense. It would just be copied over to our assembly anyway. Perhaps we should use `AppendLine()` here though, rather than passing in the newlines explicitly.

Reply:
I've created a constant called `BATCH_SEPARATOR` in this file

Reply:
Agreed about `AppendLine()`, I can't remember what I was thinking when I wrote it this way but it doesn't make much sense to me looking at it now 😄

Reply:
Done